txreviewbot/_machine.py

```diff
@@ -0,0 +1,124 @@
+import attr
+from automat import MethodicalMachine
+
+
+@attr.s
+class AddLabel(object):
+    name = attr.ib()
+
+
+@attr.s
+class RemoveLabel(object):
+    name = attr.ib()
+
+
+class ReviewMachine(object):
+    _machine = MethodicalMachine()
+
+    @_machine.input()
+    def review(self, pullRequest):
+        """
+
+        """
+
+    @_machine.input()
+    def close(self, pullRequest):
+        """
+
+        """
+
+    @_machine.input()
+    def approve(self, pullRequest):
+        """
+
+        """
+
+    @_machine.input()
+    def requestChanges(self, pullRequest):
+        """
+
+        """
+
+    @_machine.input()
+    def unsubmit(self, pullRequest):
+        """
+
+        """
+
+    @_machine.input()
+    def merge(self, pullRequest):
+        """
+
+        """
+
+    @_machine.state(initial=True)
+    def unsubmitted(self):
+        """
+        A PR that is open but not up for review.
+        """
+
+    @_machine.state()
+    def closed(self):
+        """
+
+        """
+
+    @_machine.state()
+    def needsReview(self):
+        """
+
+        """
+
+    @_machine.state()
+    def approved(self):
+        """
+
+        """
+
+    @_machine.state()
+    def changesNeeded(self):
+        """
+
+        """
+
+    @_machine.state()
+    def merged(self):
+        """
+
+        """
+
+    @_machine.output()
```

Comment:
Ah, of course!  I'm going to land this in favor of a follow-up PR that uses this approach.